### DISCUSSION TITLE
Detects a lot of Javascript instead of PHP

### DISCUSSION THREAD

**Discussion by yayimhavingfun** [github-linguist/linguist#6482] [Classification] [answered]
i just commited my repository to github and it said that it is 72.2% Javascript, which simply cannot be correct bc the project is built with Yii2

https://github.com/yayimhavingfun/taskforce <-url

last modified on 20/07/2023

should be PHP, detected as Javascript

**Comment by lildude** [comment, marked as answer]
> which simply cannot be correct bc the project is built

It certainly can. [Lingusit calculates based on the bytes of code](https://github.com/github-linguist/linguist/blob/master/docs/how-linguist-works.md) and you have considerably more Javascript than PHP:

```console
➜ github-linguist taskforce/
72.24%  685476     JavaScript
19.57%  185642     PHP
6.64%   63003      HTML
1.50%   14206      CSS
0.05%   517        Batchfile
```

These four files account for all of that:

```
JavaScript:
web/js/dropzone-amd-module.js
web/js/dropzone.js
web/js/landing.js
web/js/main.js
```

```console
➜ /bin/ls -lh taskforce/web/js/
total 1592
-rw-r--r--  1 lildude  staff   333K 21 Jul 07:39 dropzone-amd-module.js
-rw-r--r--  1 lildude  staff   333K 21 Jul 07:39 dropzone.js
-rw-r--r--  1 lildude  staff   112K 21 Jul 07:39 dropzone.min.js  # Linguist ignores this file
-rw-r--r--  1 lildude  staff   882B 21 Jul 07:39 landing.js
-rw-r--r--  1 lildude  staff   2.2K 21 Jul 07:39 main.js
```

`dropzone-amd-module.js` and `dropzone.js` are identical too and appear to be unminified versions of `dropzone.min.js` so you can easily change things by removing the duplicate and/or the unminified versions.

You can also implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to tell Linguist to ignore any of those files.

**Reply by yayimhavingfun** [reply]
oh! thank you so much, i forgot about them completely!
